api/utils/tlsutils/tlsutils.go

```diff
@@ -81,14 +81,5 @@ func TLSDial(ctx context.Context, dialer ContextDialer, network, addr string, tl
 		return nil, trace.Wrap(err)
 	}
 
-	if tlsConfig.InsecureSkipVerify {
-		return conn, nil
-	}
-
-	if err := conn.VerifyHostname(tlsConfig.ServerName); err != nil {
-		plainConn.Close()
-		return nil, trace.Wrap(err)
-	}
```

Comment:
It was never necessary, that's the same check (minus a few pieces) that's already happening as part of `HandshakeContext`; this `TLSDial` function is a copy of `tls.DialWithDialer` but with an interface ContextDialer rather than a `*net.Dialer`, and the original function doesn't do these redundant checks either: https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/crypto/tls/tls.go;l=118-164